src/link/Coff.zig

```diff
@@ -3748,4 +3748,63 @@ const Value = @import("../Value.zig");
 const AnalUnit = InternPool.AnalUnit;
 const dev = @import("../dev.zig");
 
-const msdos_stub = @embedFile("msdos-stub.bin");
+/// This is the start of a Portable Executable file.
+/// It starts with a DOS header followed by a DOS stub.
+/// This data does not change so we include it as follows in all binaries.
+///
+/// In this context,
+/// A "paragraph" is 16 bytes.
+/// A "page" is 512 bytes.
+/// A "long" is 4 bytes.
+/// A "word" is 2 bytes.
+const msdos_stub = [_]u8{
```

Comment:
style nit: `const msdos_stub: [120]u8 = .{` is better in this situation, as the length has significance (it's embedded in the stub as the "number of bytes in the last page" and "address of PE header" values).

Reply:
Can't actually do this because of the `++` and `**` etc.

Reply:
Was able to apply this by using `().*`.

Reply:
No, this keeps causing issues. Would prefer to leave this as-is for now.

Reply:
> This first initializer is only 28 bytes.

But the `[120]u8` is a type annotation on `msdos_stub`, not on the whole thing?  The annotation should work fine -- I just tested locally with a similar example.

Actually, ugh, the `++` with strings makes things weird... god I hate that operator, it results in random surprising types. `msdos_stub` here is actually a slice rather than an array. But it should work fine to annotate `: [120]u8` and do `.*` on the strings so that everything is an array -- again, testing locally, this seems to work.

Reply:
> But it should work fine to annotate : [120]u8 and do .* on the strings so that everything is an array -- again, testing locally, this seems to work.

IIRC this is what I tried but the CI failed (missing `&` operators where `msdos_stub` is used) but I might've done something slightly different so I'll try one more time.

Reply:
Oh right, yes, the use cases might be depending on it being a slice. If you prefer to keep it a slice, would you mind annotating `msdos_stub: []const u8`, just to make that clear?

Reply:
Yeah, this causes a difference.

Currently:
```
@as(type, *const [120:0]u8)
```
After Changing to `: [120]u8 = .{` and adding `.*` to the end of the one string literal below:
```
Compile Log Output:
@as(type, [120]u8)
```

Reply:
It's just going to be one additional `&` actually so I think this would be fine.

Reply:
It can be a slice or array pointer if you prefer, just if so I'd still like the type to be annotated, since it's pretty confusing when `++` is involved :P

Reply:
Yup, annotated that now.